include/materials/NuclearMaterial.h

```diff
@@ -0,0 +1,123 @@
+ *                  xsec(T) = a * T + b
+ */
+class NuclearMaterial : public GenericConstantMaterial
+{
+public:
+  NuclearMaterial(const InputParameters & parameters);
+
+  // correspond to descriptions above
+  enum INTERPOLATOR
+  {
+    BICUBIC,
+    SPLINE,
+    MONOTONE_CUBIC,
+    LINEAR,
+    NONE,
+    LSQ
+  };
+
+  // returns a MooseEnum corresponding to the above enum
+  static MooseEnum interpTypes()
+  {
+    return MooseEnum("bicubic=0 spline=1"
+                     " monotone_cubic=2 linear=3 none=4 least_squares=5");
+  }
+
+protected:
+  virtual void dummyComputeQpProperties();
+  virtual void splineComputeQpProperties();
+  virtual void monotoneCubicComputeQpProperties();
+  virtual void linearComputeQpProperties();
+
+  const VariableValue & _temperature;
+
+  MaterialProperty<std::vector<Real>> & _remxs;
+  MaterialProperty<std::vector<Real>> & _fissxs;
+  MaterialProperty<std::vector<Real>> & _nsf;
+  MaterialProperty<std::vector<Real>> & _fisse;
+  MaterialProperty<std::vector<Real>> & _diffcoef;
+  MaterialProperty<std::vector<Real>> & _recipvel;
+  MaterialProperty<std::vector<Real>> & _chi;
+  MaterialProperty<std::vector<Real>> & _chi_d;
+  MaterialProperty<std::vector<Real>> & _gtransfxs;
+  MaterialProperty<std::vector<Real>> & _beta_eff;
+  MaterialProperty<Real> & _beta;
+  MaterialProperty<std::vector<Real>> & _decay_constant;
+  MaterialProperty<std::vector<Real>> & _d_remxs_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_fissxs_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_nsf_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_fisse_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_diffcoef_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_recipvel_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_chi_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_chi_d_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_gtransfxs_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_beta_eff_d_temp;
+  MaterialProperty<Real> & _d_beta_d_temp;
+  MaterialProperty<std::vector<Real>> & _d_decay_constant_d_temp;
+
+  MooseEnum _interp_type;
+
+  unsigned _num_groups;
+  unsigned _num_precursor_groups;
+  std::map<std::string, std::vector<Real>> _xsec_map;
+  std::map<std::string, std::vector<SplineInterpolation>> _xsec_spline_interpolators;
+  std::map<std::string, std::vector<MonotoneCubicInterpolation>> _xsec_monotone_cubic_interpolators;
+  std::map<std::string, std::vector<LinearInterpolation>> _xsec_linear_interpolators;
+  std::map<std::string, std::vector<BicubicSplineInterpolation>> _xsec_bicubic_spline_interpolators;
+  std::map<std::string, int> _vec_lengths;
+  std::map<std::string, std::string> _file_map;
+  std::vector<double> _XsTemperature;
+
+  std::vector<std::vector<Real>> _flux_consts = std::vector<std::vector<Real>>(2);
+  std::vector<std::vector<Real>> _remxs_consts = std::vector<std::vector<Real>>(2);
```

Comment:
By the way, why is it that we initialize these to two vectors again? You may not be able to initialize them once this is converted to an abstract base class. If it is always two vectors, maybe an std::pair would be more expressive.